client/src/components/content/RobotContainer/RobotContainer.jsx

```diff
@@ -54,31 +92,62 @@ const RobotContainer = (props) => {
       {hoveredUpon && (
         <Col className={[styles.box, styles.robotBox, styles.selectedRobotBox]}>
           <Row align='middle' style={{ height: '55%' }}>
-            <Col type='flex' span={12}>
+            <Col type='flex' span={8}>
               <Link
                 to={{
                   pathname: `/interaction_cockpit/${robotId}`,
                   state: { userId },
                 }}
               >
-                <PlayCircleOutlined
-                  onClick={initLocalSsot}
-                  className={styles.clickableIcon}
-                />
+                <Tooltip title='Execute Robot'>
+                  <PlayCircleOutlined
+                    onClick={initLocalSsot}
+                    className={styles.clickableIcon}
+                  />
+                </Tooltip>
               </Link>
             </Col>
-            <Col type='flex' span={12}>
+            <Col type='flex' span={8}>
               <Link to={`/modeler/${robotId}`}>
-                <EditOutlined className={styles.clickableIcon} />
+                <Tooltip title='Edit Robot'>
+                  <EditOutlined className={styles.clickableIcon} />
+                </Tooltip>
               </Link>
             </Col>
+            <Col type='flex' span={8}>
+              <Popconfirm
+                title='Delete Robot?'
+                visible={popConfirmVisible}
+                onConfirm={deleteRobot}
+                okButtonProps={{ loading: confirmLoading }}
+                onCancel={handlePopConfirmCancel}
+                okText='Delete'
+                icon={<WarningOutlined />}
+                placement="left"
+              >
```

Comment:
Just a consideration for a better readability. The render method has become extremely long. Normally we would dump individual components into new `.jsx` files. I don't think that makes much sense here.
What do you think about using new render-functions for those React-Components like the Popconfirm or general the row with the three icons.